packages/contracts/contracts/ChugSplashRegistry.sol

```diff
@@ -53,23 +67,48 @@ contract ChugSplashRegistry {
      */
     mapping(ChugSplashManager => bool) public managers;
 
+    mapping(bytes32 => address) public adapters;
+
     /**
      * @notice Registers a new project.
      *
-     * @param _name  Name of the new ChugSplash project.
-     * @param _owner Initial owner for the new project.
+     * @param _name         Name of the new ChugSplash project.
+     * @param _owner        Initial owner for the new project.
+     * @param _proxyUpdater Address of the ProxyUpdater for this project. Projects that do not use a custom
+     *                      proxy should set this to the zero-address.
+     * @param _proxyAdmin   Address of the ProxyAdmin for this project. Projects that do not use a custom
+     *                      proxy should set this to the zero-address.
      */
-    function register(string memory _name, address _owner) public {
+    function register(
+        string memory _name,
+        address _owner,
+        address _proxyUpdater,
```

Comment:
We can use the same `ProxyUpdater` implementation for every contract. I would recommend making this an `immutable` that either gets passed into the constructor of the `ChugSplashRegistry`. We then don't need users to pass the value in here themselves. We can then either also pass it into the constructor of the `ChugSplashManager` or we can simply call back to the registry from inside the manager. Calling back will be slightly more expensive.